### DISCUSSION TITLE
Accessing client IP address in SvelteKit?

### DISCUSSION THREAD

**Discussion by hyunbinseo** [sveltejs/kit#3973] [Q&A] [answered]
Is there a way to access client IP address in SvelteKit?

---

**Resources**

- [Can I get the IP address of my users in a SvelteKit project?](https://stackoverflow.com/questions/70512230/)
- https://github.com/sveltejs/kit/pull/3972
- https://github.com/sveltejs/kit/pull/3993

> And, while `req.socket.remoteAddress` may be available in Node environments, it may not be in other environments that use this same utility function. This feels like a job for [each relevant adapter](https://kit.svelte.dev/docs/adapters#supported-environments-platform-specific-context).

**Comment by hyunbinseo** [comment]
Planning to use an external IP address API in the mean time.

Idea from [`byronbyron/ip-address-tracker`](https://github.com/byronbyron/ip-address-tracker) project, which uses `SvelteKit` and `ipify`.

List of 3rd party services, and their limitations are organized in a StackOverflow [question](https://stackoverflow.com/questions/391979/how-to-get-clients-ip-address-using-javascript).

Personally decided to implement a custom [Cloudflare Workers](https://workers.cloudflare.com/).

**Reply by evdama** [reply]
here's the current issue https://github.com/sveltejs/kit/pull/3993

**Comment by hyunbinseo** [comment, marked as answer]
Resolved with https://github.com/sveltejs/kit/pull/4289

**Comment by CAPTAIN320** [comment]
I made an example of how to get a user's ip address and subsequently the user's location using sveltekit.

Repo - https://github.com/CAPTAIN320/sveltekit-user-ip-location-example
